diff --git a/00_static.html b/01_declarative.php
index c89069f..185c35d 100644
--- a/00_static.html
+++ b/01_declarative.php
@@ -1,10 +1,14 @@
-<ul>
-  <li class="first">First Content</li>
-  <li>Some Content</li>
-  <li>Some Content</li>
-  <li>Some Content</li>
-  <li>Some Content</li>
-  <li>Some Content</li>
-  <li>Some Content</li>
-  <li class="last">Last Content</li>
-</ul>
+<?php
+
+echo "<ul>";
+echo "  <li class=\"first\">First Content</li>";
+echo "  <li>Some Content</li>";
+echo "  <li>Some Content</li>";
+echo "  <li>Some Content</li>";
+echo "  <li>Some Content</li>";
+echo "  <li>Some Content</li>";
+echo "  <li>Some Content</li>";
+echo "  <li class=\"last\">Last Content</li>";
+echo "</ul>";
+
+?>
