diff --git a/02_procedural.php b/03_objectoriented.php
index 644ab2a..88afa60 100644
--- a/02_procedural.php
+++ b/03_objectoriented.php
@@ -1,63 +1,47 @@
 <?php
 
-/**
- * ============================================================================
- * Programming before loops
- * ============================================================================
- */
-
-echo "<ul>";
-echo "  <li class=\"first\">First Content</li>";
-
-
-$i = 0;
-
-echoline:
-echo "  <li>Some Content</li>";
-$i = $i + 1;
-
-if ($i < 6) goto echoline;
-
-
-echo "  <li class=\"last\">Last Content</li>";
-echo "</ul>";
-
-/**
- * ============================================================================
- * Programming with loops, w/o functions
- * ============================================================================
- */
-
-echo "<ul>";
-echo "  <li class=\"first\">First Content</li>";
-for ($i = 0; $i < 6; $i++) {
-  echo "  <li>Some Content</li>";
-}
-echo "  <li class=\"last\">Last Content</li>";
-echo "</ul>";
-
-/**
- * ============================================================================
- * Programming with functions
- * ============================================================================
- */
-
-function echo_li($content, $class = FALSE) {
-  $li = ($class) ? "<li class=\"" . $class . "\">" : "<li>";
-  $li .= $content . "</li>";
-  echo $li;
+class listClass {
+  var $name;
+  var $items;
+
+  function addItem($content, $class = FALSE) {
+    $this->items[] = array(
+      'content' => $content,
+      'class' => $class,
+    );
+  }
+
+  function printList() {
+    echo "<ul>";
+    echo "<h3>" . $this->name . "</h3>";
+
+    foreach ($this->items as $item) {
+      $li = ($item['class']) ? "<li class=\"" . $item['class'] . "\">" : "<li>";
+      $li .= $item['content'] . "</li>";
+      echo $li;
+    }
+
+    echo "</ul>";
+  }
 }
 
-echo "<ul>";
+function makeList($name) {
+  $list = new listClass;
+  $list->name = $name;
 
-echo_li('First Content', 'first');
+  $list->addItem('First Content', 'first');
+  for ($i = 0; $i < 6; $i++) {
+    $list->addItem('Some Content');
+  }
+  $list->addItem('Last Content', 'last');
 
-for ($i = 0; $i < 6; $i++) {
-  echo_li('Some Content');
+  return $list;
 }
 
-echo_li('Last Content', 'last');
+$firstList = makeList('My first list object');
+$secondList = makeList('My second list object');
 
-echo "</ul>";
+$firstList->printList();
+$secondList->printList();
 
 ?>
