diff --git a/06_inheritance.php b/07_pubprivprot.php
index f6e98f5..4fb1402 100644
--- a/06_inheritance.php
+++ b/07_pubprivprot.php
@@ -1,9 +1,9 @@
 <?php
 
 class element {
-  var $type;
-  var $content; // array of content, either string or element.
-  var $attr; // array of arrays, keyed by attribute name (e.g. array('class' => array('one', 'two'))
+  protected $type;
+  protected $content; // array of content, either string or element.
+  protected $attr; // array of arrays, keyed by attribute name (e.g. array('class' => array('one', 'two'))
 
   function __construct($type, $content = NULL, $attr = NULL) {
     $this->type = $type;
@@ -49,7 +49,7 @@ class li extends element {
 }
 
 class ul extends element {
-  var $name;
+  public $name;
 
   function __construct($name, $attr = NULL) {
     $this->name = $name;
@@ -66,7 +66,7 @@ class ul extends element {
     $this->content[] = $item;
   }
 
-  function makeExample() {
+  private function makeExample() {
     $item = new li('First Content', array('class' => array('first')));
     $this->addItem($item);
 
@@ -86,4 +86,8 @@ $secondList = new ul('My second list object');
 $firstList->printElem();
 $secondList->printElem();
 
+$firstLIst->name = 'My first list with a new name';
+echo $firstLIst->name;
+$firstList->printElem(); // Why does this not print the new name?
+
 ?>
